### ISSUE TITLE
FlxTween.quadPath() can make the object disappear from the screen

### ISSUE THREAD

**Issue by giuppe** [HaxeFlixel/flixel#2958]
- **Haxe version:** 4.3.2
- **Flixel version:** 5.5.0
- **OpenFL version:** ?.?.?
- **Lime version:** ?.?.?
- **Affected targets:** All

There is a condition where `FlxTween.quadPath()` will make the FlxObject disappear from the screen.
________________________________

**Code snippet reproducing the issue**:

```haxe
package;

import flixel.FlxState;

class PlayState extends FlxState
{
	override public function create():Void
	{
		_img = new FlxSprite(0, 0);
		_img.makeGraphic(100, 100, FlxColor.MAGENTA);
		add(_img);
		var point1 = FlxPoint.get(100,100);
		var point2 = FlxPoint.get(100,100);
		var point3 = FlxPoint.get(0,100);
		var point4 = FlxPoint.get(100,0);
		var point5 = FlxPoint.get(200,200);
		FlxTween.quadPath(_img, [point1, point2, point3, point4, point5], 10);
	}
}
```

The position of the FlxObject is wrongly set to (NaN, NaN).

Requirements for the bug to happen:
- at least 5 points;
- two subsequent points in the same location.

Obviously, nobody would manually set the points in this way, but it can happen if they are generated by a game mechanic.

The culprit seems to be this line of QuadPath: https://github.com/HaxeFlixel/flixel/blob/c0a6f1c7fc7fc155641853968127a113a5b6b729/flixel/tweens/motion/QuadPath.hx#L221

When both `ba` and `c2` are zero, there is a division by zero that makes the function return NaN; that then propagates through all the other computations until it ends on the FlxObject position.
`ba == c2 == 0` when the first two arguments of the functions are the same point.

I guess the fastest way to avoid that discontinuity is to set `(ba + c2)` to a small quantity (for example 0.00001) instead of zero - it would make the division by zero disappear.

I can produce a PR if needed.

________________________________

**Observed behavior:** No square on the screen.

**Expected behavior:** Square on the screen moving in a curve from `point1` to `point5`.

**Comment by Geokureli** [comment]
feel free to make a PR

**Comment by giuppe** [comment]
PR produced: #3019

**Comment by Geokureli** [comment]
Fixed via #3019
